hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -513,16 +574,34 @@ public SerializerAdapter serializerFor(final Object object) {
             serializer = lookupGlobalSerializer(type);
         }
 
+        //6-A good candidate to be serialized via compact serializer
         if (serializer == null) {
-            if (active) {
-                throw new HazelcastSerializationException("There is no suitable serializer for " + type);
-            }
-            throw notActiveExceptionSupplier.get();
+            serializer = lookupCompactSerializer(type, includeSchema);
         }
+
+        if (serializer == null) {
+            throw new HazelcastSerializationException("There is no suitable serializer for " + type);
+        }
+
         return serializer;
     }
 
-    private SerializerAdapter lookupDefaultSerializer(Class type) {
+    private SerializerAdapter lookupCompactSerializer(Class type, boolean includeSchema) {
+        if (type.getName().startsWith("java.")) {
```

Comment:
I don't want to direct JDK classes to the compactSerializer. But now thinking about it, this is a heavy operation on the hotpath. And the Serializable jdk classes will be handled before this line anyway. I will remove the check because of performance considerations.